engine/apps/mattermost/tests/test_mattermost_channel.py

```diff
@@ -0,0 +1,351 @@
+
+    assert response.status_code == expected_status
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "role,expected_status",
+    [
+        (LegacyAccessControlRole.ADMIN, status.HTTP_200_OK),
+        (LegacyAccessControlRole.EDITOR, status.HTTP_200_OK),
+        (LegacyAccessControlRole.VIEWER, status.HTTP_200_OK),
+        (LegacyAccessControlRole.NONE, status.HTTP_403_FORBIDDEN),
+    ],
+)
+def test_get_mattermost_channels_permissions(
+    make_organization_and_user_with_plugin_token,
+    make_user_auth_headers,
+    make_mattermost_channel,
+    role,
+    expected_status,
+):
+    client = APIClient()
+    organization, user, token = make_organization_and_user_with_plugin_token(role)
+    mattermost_channel = make_mattermost_channel(organization=organization)
+
+    url = reverse("mattermost:channel-detail", kwargs={"pk": mattermost_channel.public_primary_key})
+    response = client.get(url, **make_user_auth_headers(user, token))
+
+    assert response.status_code == expected_status
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "role,expected_status",
+    [
+        (LegacyAccessControlRole.ADMIN, status.HTTP_204_NO_CONTENT),
+        (LegacyAccessControlRole.EDITOR, status.HTTP_403_FORBIDDEN),
+        (LegacyAccessControlRole.VIEWER, status.HTTP_403_FORBIDDEN),
+        (LegacyAccessControlRole.NONE, status.HTTP_403_FORBIDDEN),
+    ],
+)
+def test_delete_mattermost_channels_permissions(
+    make_organization_and_user_with_plugin_token,
+    make_user_auth_headers,
+    make_mattermost_channel,
+    role,
+    expected_status,
+):
+    client = APIClient()
+    organization, user, token = make_organization_and_user_with_plugin_token(role)
+    mattermost_channel = make_mattermost_channel(organization=organization)
+
+    url = reverse("mattermost:channel-detail", kwargs={"pk": mattermost_channel.public_primary_key})
+    response = client.delete(url, **make_user_auth_headers(user, token))
+
+    assert response.status_code == expected_status
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "role,expected_status",
+    [
+        (LegacyAccessControlRole.ADMIN, status.HTTP_201_CREATED),
+        (LegacyAccessControlRole.EDITOR, status.HTTP_403_FORBIDDEN),
+        (LegacyAccessControlRole.VIEWER, status.HTTP_403_FORBIDDEN),
+        (LegacyAccessControlRole.NONE, status.HTTP_403_FORBIDDEN),
+    ],
+)
+def test_post_mattermost_channels_permissions(
+    make_organization_and_user_with_plugin_token,
+    make_mattermost_get_channel_by_name_team_name_response,
+    make_user_auth_headers,
+    role,
+    expected_status,
+):
+    client = APIClient()
+    _, user, token = make_organization_and_user_with_plugin_token(role)
+
+    data = make_mattermost_get_channel_by_name_team_name_response()
+    channel_response = requests.Response()
+    channel_response.status_code = status.HTTP_200_OK
+    channel_response._content = json.dumps(data).encode()
```

Comment:
No need to change but, FYI, we are using [httpretty](https://github.com/grafana/oncall/blob/dev/engine/apps/webhooks/tests/test_trigger_webhook.py#L236) for tests involving http requests